src/tribler-common/tribler_common/sentry_reporter/sentry_reporter.py

```diff
@@ -33,39 +33,58 @@
 RELEASE = 'release'
 
 
-class SentryReporter:
-    """SentryReporter designed for sending reports to the Sentry server from
-    a Tribler Client.
+class SentryStrategy(Enum):
+    """Class describes all available Sentry Strategies
 
-    It can work with 3 strategies:
+    SentryReporter can work with 3 strategies:
     1. Send reports are allowed
     2. Send reports are allowed with a confirmation dialog
-    3. Send reports are suppressed (disallowed, but the last event will be stored)
+    3. Send reports are suppressed (but the last event will be stored)
+    """
+
+    SEND_ALLOWED = auto()
+    SEND_ALLOWED_WITH_CONFIRMATION = auto()
+    SEND_SUPPRESSED = auto()
+
+
+class ThisSentryStrategy:
+    """Class simplify execution of a particular block of code with given SentryStrategy
```

Comment:
You can write this context manager as a generator, this way it is a bit easier to understand how it handles exceptions:

```python
from contextlib import contextmanager


@contextmanager
def this_sentry_strategy(strategy: SentryStrategy):
    saved_strategy = SentryReporter._context_strategy.get()
    try:
        SentryReporter._context_strategy.set(strategy)
        yield
    finally:
        SentryReporter._context_strategy.set(saved_strategy)
```

Reply:
Wow, cool, thank you!